src/components/partner/partner.service.ts

```diff
@@ -189,9 +189,48 @@ export class PartnerService {
     { filter, ...input }: PartnerListInput,
     session: Session
   ): Promise<PartnerListOutput> {
-    const query = this.repo.list({ filter, ...input }, session);
+    if (await this.authorizationService.canList(Partner, session)) {
+      const query = this.repo.list({ filter, ...input }, session);
+      return await runListQuery(query, input, (id) =>
+        this.readOne(id, session)
+      );
+    } else {
+      const listPartnerIds = [] as ID[];
+      const result = await this.repo
+        .listPartnersOfAllUserProjects(session, { filter, ...input })
+        .run();
 
-    return await runListQuery(query, input, (id) => this.readOne(id, session));
+      // I have to put an await, because the next block will execute before this one is finished....
+      // eslint-disable-next-line @typescript-eslint/await-thenable, @typescript-eslint/no-misused-promises
+      await void result.forEach(async (project) => {
+        // eslint-disable-next-line @typescript-eslint/await-thenable, @typescript-eslint/no-misused-promises, @typescript-eslint/no-confusing-void-expression
+        await project.resources.forEach(async (resource) => {
```

Comment:
The linter is yelling at you because this doesn't work. You'll get passed line 220 before line 215 ever executes. 

Simplist option is to switch to a for-of-await syntax.
```ts
for await (const project of result) {
  for await (const resource of project.resources) {
    await canList()
  }
}
```